src/resources/app/pages/course/CreateCourseMaterial.vue

```diff
@@ -122,10 +145,27 @@ export default {
         }
         this.$inertia.visit(`/course/${this.$props.id}/classwork`);
       } catch (error) {
-        this.$store.dispatch("OPEN_SNACKBAR", "Error uploading data");
+        await this.$store.dispatch("OPEN_SNACKBAR", "Error uploading data");
       }
       this.isSubmitting = false;
     },
+    async checkTeacher(courseId) {
+      try {
+        const response = await courseDetail(courseId);
+        if (response.data.createdBy !== this.$store.state.user.id){
+          window.location.assign("/404");
```

Comment:
using `window.location.assign()` will force the page to refresh, use `this.$inertia.visit()` instead for better user experience